src/ansys/additive/core/microstructure_3d.py

```diff
@@ -0,0 +1,397 @@
+        )
+        self._sample_min_x = value
+
+    @property
+    def sample_min_y(self) -> float:
+        """Minimum y coordinate of the geometry sample (m).
+
+        Valid values are from the :obj:`MIN_POSITION_COORDINATE` value to the
+        :obj:`MAX_POSITION_COORDINATE` value.
+        """
+        return self._sample_min_y
+
+    @sample_min_y.setter
+    def sample_min_y(self, value: float):
+        self.__validate_range(
+            value, self.MIN_POSITION_COORDINATE, self.MAX_POSITION_COORDINATE, "sample_min_y"
+        )
+        self._sample_min_y = value
+
+    @property
+    def sample_min_z(self) -> float:
+        """Minimum z coordinate of the geometry sample (m).
+
+        Valid values are from the :obj:`MIN_POSITION_COORDINATE` value to the
+        :obj:`MAX_POSITION_COORDINATE` value.
+        """
+        return self._sample_min_z
+
+    @sample_min_z.setter
+    def sample_min_z(self, value: float):
+        self.__validate_range(
+            value, self.MIN_POSITION_COORDINATE, self.MAX_POSITION_COORDINATE, "sample_min_z"
+        )
+        self._sample_min_z = value
+
+    @property
+    def sample_size_x(self) -> float:
+        """Size of the geometry sample in the x direction (m).
+
+        Valid values are from the :obj:`MIN_SAMPLE_SIZE` value to the
+        :obj:`MAX_SAMPLE_SIZE` value.
+        """
+        return self._sample_size_x
+
+    @sample_size_x.setter
+    def sample_size_x(self, value: float):
+        self.__validate_range(value, self.MIN_SAMPLE_SIZE, self.MAX_SAMPLE_SIZE, "sample_size_x")
+        self._sample_size_x = value
+
+    @property
+    def sample_size_y(self) -> float:
+        """Size of the geometry sample in the y direction (m).
+
+        Valid values are from the :obj:`MIN_SAMPLE_SIZE` value to the
+        :obj:`MAX_SAMPLE_SIZE` value.
+        """
+        return self._sample_size_y
+
+    @sample_size_y.setter
+    def sample_size_y(self, value: float):
+        self.__validate_range(value, self.MIN_SAMPLE_SIZE, self.MAX_SAMPLE_SIZE, "sample_size_y")
+        self._sample_size_y = value
+
+    @property
+    def sample_size_z(self) -> float:
+        """Size of the geometry sample in the z direction (m).
+
+        Valid values are from the :obj:`MIN_SAMPLE_SIZE` value to the
+        :obj:`MAX_SAMPLE_SIZE` value.
+        """
+        return self._sample_size_z
+
+    @sample_size_z.setter
+    def sample_size_z(self, value: float):
+        self.__validate_range(value, self.MIN_SAMPLE_SIZE, self.MAX_SAMPLE_SIZE, "sample_size_z")
+        self._sample_size_z = value
+
+    @property
+    def calculate_initial_microstructure(self) -> bool:
+        """Whether to run the initial microstructure solver.
+
+        If ``True``, the initial microstructure solver will be used to create initial
+        condition grain identifiers and euler angles.
+        If ``False``, the initial microstructure solver will not be run.
```

Comment:
```suggestion
        If ``True``, the initial microstructure solver is used to create initial
        condition grain identifiers and Euler angles.
        If ``False``, the initial microstructure solver is not be run.
```
Avoid future tense. A quick Google search showed that "Euler" is usually treated as a proper noun.